assets/js/googlesitekit/api/cache.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * External dependencies
+ */
+import { isEqual } from 'lodash';
+
+let storageBackend;
+/**
+ * Override the storage backend.
+ *
+ * Largely used for tests. Should not be used directly.
+ *
+ * @param {*} backend Backend to set for the cache.
+ */
+export const _setSelectedStorageBackend = ( backend ) => {
```

Comment:
Let's refrain from using underscore-prefixed functions to indicate they're private. If we specifically want to annotate a function as private, it should receive `@access private`.

Also, "private" functions like this should not be exported. In this case, the function is currently only used in the same file, so no purpose to export it anyway.

Reply:
Given that the functions using this are unnecessary for now, let's remove this function as well.

Reply:
Because it makes sense for some private functions to exported for testing purposes, I wanted to be extra-clear about the intended usage of these functions. I'll add the `@access` annotations where appropriate as well, but I think marking functions as private but exporting them for either internal or testing use is valuable. We can communicate that an API is not for public use, but having these available for testing makes our tests much better.

Reply:
I think this is worth discussing when we need them, but I don't see a compelling reason for these functions yet, most of them are actually unused. Let's add functions only when we need them so we can discuss with context.

Reply:
These functions are used to ensure each storage mechanism works properly in the tests. I suppose I could mock out the globals for each run instead (`global.localStorage` and `global.sessionStorage`), which _technically_ more fully approximates how the mechanism is chosen and selected, but it's a messier set of mocks.

These functions all exist to make the tests work and are exported simply because we _need_ to export them to test the code. They aren't for public consumption though, and I think it's fine to export private APIs for testing but make it clear they're entirely unsupported in the docs.

Reply:
(You can see their usage in https://github.com/google/site-kit-wp/blob/ccb9212a9cdee2d6ad41ccf779b152990368d936/assets/js/googlesitekit/api/cache.test.js#L6-L10, where ESLint would have complained if they were unused.)